### PR TITLE
[CORL-90] moderator notes

### PR DESCRIPTION
**Pull request by tessalt**
Adds a `moderatorNotes` field to `User` and interface for creating and deleting notes in user drawer. 

Screen Shot 2019-09-30 at 9 05 57 AM
Screen Shot 2019-09-30 at 9 08 56 AM

### CODE REVIEW

**Review by nick-funk — COMMENTED** [review]
Went through and code changes look good.

Looks like a migration is needed, assuming that was held off as we need @wyattjoh 's migration PR to be merged in first. Will hold off further testing until then.

**Comment by tessalt** [comment]
@nick-funk @wyattjoh added migration

File: src/core/client/admin/components/UserHistoryDrawer/UserDrawerNotesContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  );
+  const onSubmit = useCallback(
+    async ({ body }, form: FormApi) => {
+      await createNote({
+        userID: user.id,
+        body,
+      });
+      form.reset();
+    },
+    [createNote, user]
```
Comment (wyattjoh) [inline src/core/client/admin/components/UserHistoryDrawer/UserDrawerNotesContainer.tsx]: You can omit the `createNote` here.

File: src/core/server/models/user/user.ts [hunk 2]
```diff
@@ -2323,3 +2351,78 @@ export async function retrieveUserScheduledForDeletion(
...
+    },
+    {
+      $pull: {
+        moderatorNotes: { id, createdBy },
+      },
+    },
+    {
+      returnOriginal: false,
+    }
+  );
```
Comment (wyattjoh) [inline src/core/server/models/user/user.ts:2426]: Probably good to add the guard clause from the `createModeratorNote` function that returns the `UserNotFoundError` on the falsy response of `findOneAndUpdate`.

File: src/core/server/services/migrate/migrations/1569947670260_add_moderator_notes_to_user.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      },
+      "added empty moderatorNotes array"
+    );
+  }
+
+  public async down(mongo: Db, tenantID: string) {
+    const result = await collections.users(mongo).updateMany(
+      {
+        tenantID,
+        $exists: "moderatorNotes",
```
Comment (wyattjoh) [inline src/core/server/services/migrate/migrations/1569947670260_add_moderator_notes_to_user.ts]: The syntax for the `$exists` operator is actually:

```suggestion
        moderatorNotes: { $exists: true },
```

**Review by wyattjoh — CHANGES_REQUESTED** [review]

**Comment by tessalt** [comment]
updated

File: src/core/client/admin/components/UserHistoryDrawer/UserDrawerNotesContainer.css [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  padding: calc(0.5 * var(--mini-unit));
+  border-radius: 2px;
+  margin-bottom: var(--spacing-2);
+  padding: var(--spacing-3);
+  font-weight: var(--font-weight-regular);
+  font-family: var(--font-family-sans-serif);
+  font-size: calc(16rem / var(--rem-base));
+  line-height: 1;
+  letter-spacing: calc(0.2em / 16);
+  color: var(--palette-text-primary);
```
Comment (wyattjoh) [inline src/core/client/admin/components/UserHistoryDrawer/UserDrawerNotesContainer.css:20]: Firefox adds a `border-width: 2px;` when unspecified, perhaps add a `border-width: 1px;` to have it match chrome?

**Review by wyattjoh — APPROVED** [review]

**Pull request merged**